src/Arcus.Security.Core/Caching/CachedSecretProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Arcus.Security.Core.Exceptions;
+using Arcus.Security.Core.Interfaces;
+using GuardNet;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Arcus.Security.Core.Caching
+{
+    /// <inheritdoc cref="ISecretProvider"/>
+    /// <summary>
+    /// A Secret Provider that will cache secrets in memory, to improve performance
+    /// </summary>
+    public class CachedSecretProvider : ISecretProvider
```

Comment:
This should implement ICachedSecretProvider that enforces Get(name, skipCache:false) to force fresh gets in case of rolled secrets

Reply:
ok

Reply:
Feel free to have a look at my key Vault rolling sample repo on Github